File: app/src/main/java/com/example/edt73/MainActivity.java
package com.example.edt73;

import androidx.appcompat.app.AppCompatActivity;

import android.os.Bundle;
import android.view.View;
import android.view.textclassifier.TextLanguage;
import android.widget.Button;
import android.widget.LinearLayout;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;
import android.widget.Toast;

import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {
    private RadioGroup grupoTop;
    private RadioButton fatRadio;
    private RadioButton vaelRadio;
    private RadioButton lorentzRadio;
    private RadioButton creffRadio;

    private RadioGroup grupoGenero;
    private RadioButton maleRadio;
    private RadioButton femaleRadio;

    private RadioGroup grupo2;
    private RadioButton smallRadio;
    private RadioButton mediumRadio;
    private RadioButton broadRadio;

    private LinearLayout heightV1;
    private TextInputEditText height1Numero;
    private LinearLayout heightV2;
    private TextInputEditText height2Numero;
    private LinearLayout waist;
    private TextInputEditText waistNumero;
    private LinearLayout age;
    private TextInputEditText ageNumero;

    private TextView resultGrande;
    private LinearLayout resultGrandeGraella;

    private Button boton;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        grupoTop = findViewById(R.id.eleccionSuperior);
        fatRadio = findViewById(R.id.fatRadio);
        vaelRadio = findViewById(R.id.WanRadio);
        lorentzRadio = findViewById(R.id.LorentzRadio);
        creffRadio = findViewById(R.id.creffRadio);

        grupoGenero = findViewById(R.id.genero);
        maleRadio = findViewById(R.id.maleRadio);
        femaleRadio = findViewById(R.id.femaleRadio);

        grupo2 = findViewById(R.id.radio2);
        smallRadio = findViewById(R.id.smallRadio);
        mediumRadio = findViewById(R.id.mediumRadio);
        broadRadio = findViewById(R.id.broadRadio);

        heightV1 = findViewById(R.id.heightLayoutv1);
        heightV2 = findViewById(R.id.heightLayoutv2);
        waist = findViewById(R.id.waistLayout);
        age = findViewById(R.id.ageLinear);
        resultGrande = findViewById(R.id.resultadoGrande);
        resultGrandeGraella = findViewById(R.id.graella);

        height1Numero = findViewById(R.id.height11Numero);
        height2Numero = findViewById(R.id.height2Numero);
        waistNumero = findViewById(R.id.waistNumero);
        ageNumero = findViewById(R.id.ageNumero);

        maleRadio=findViewById(R.id.maleRadio);
        smallRadio=findViewById(R.id.smallRadio);

        boton = findViewById(R.id.boton);


        heightV1.setVisibility(View.INVISIBLE);
        waist.setVisibility(View.INVISIBLE);
        heightV2.setVisibility(View.INVISIBLE);
        age.setVisibility(View.INVISIBLE);


        grupoTop.setOnCheckedChangeListener(new RadioGroup.OnCheckedChangeListener() {
            @Override
            public void onCheckedChanged(RadioGroup group, int checkedId) {

                int seleccionado = grupoTop.getCheckedRadioButtonId();//esto devuelve el id del boton seleccionad
                RadioButton seleccionTexto = findViewById(seleccionado);


                switch (seleccionado) {
                    case R.id.fatRadio://el id que devuelve es el nombre pero en int
                        heightV1.setVisibility(View.VISIBLE);
                        height1Numero.setClickable(true);
                        height1Numero.setEnabled(true);

                        waist.setVisibility(View.VISIBLE);
                        waistNumero.setClickable(true);
                        waistNumero.setEnabled(true);


                        heightV2.setVisibility(View.INVISIBLE);
                        height2Numero.setClickable(false);
                        height2Numero.setEnabled(false);

                        age.setVisibility(View.INVISIBLE);
                        ageNumero.setClickable(false);
                        ageNumero.setEnabled(false);


                        grupo2.setClickable(false);
                        grupoGenero.setClickable(true);
                        maleRadio.setChecked(true);



                        boton.setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {

                                String altura = height1Numero.getEditableText().toString().trim();
                                String anchura = waistNumero.getEditableText().toString().trim();

                                if (height1Numero.getEditableText().toString().trim().isEmpty() || waistNumero.getEditableText().toString().trim().isEmpty()) {
                                    //vacio

                                    resultGrande.setText(R.string.falta);
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else if (Integer.parseInt(altura) < 0 || Integer.parseInt(anchura) < 0) {//negativo1
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrande.setText(R.string.negativo);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else {


                                    if (!genero().equals(null)) {
                                        double resultado = BodyMass.FatMass(Integer.parseInt(altura), Integer.parseInt(anchura), genero());
                                        resultGrande.setText(resultado + " %");
                                        resultGrande.setVisibility(View.VISIBLE);
                                        resultGrandeGraella.setVisibility(View.VISIBLE);
                                    }


                                }


                            }

                        });//boton


                        break;

                    case R.id.WanRadio://el id que devuelve es el nombre pero en int
                        grupoGenero.clearCheck();

                        heightV1.setVisibility(View.VISIBLE);
                        height1Numero.setClickable(true);
                        height1Numero.setEnabled(true);


                        waist.setVisibility(View.INVISIBLE);
                        waistNumero.setClickable(false);
                        waistNumero.setEnabled(false);

                        heightV2.setVisibility(View.INVISIBLE);
                        height2Numero.setClickable(false);
                        height2Numero.setEnabled(false);

                        age.setVisibility(View.INVISIBLE);
                        ageNumero.setClickable(false);
                        ageNumero.setEnabled(false);

                        resultGrande.setVisibility(View.INVISIBLE);
                        resultGrandeGraella.setVisibility(View.INVISIBLE);

                        grupo2.setClickable(false);
                        grupoGenero.setClickable(true);
                        maleRadio.setChecked(true);

                        boton.setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {

                                String altura = height1Numero.getEditableText().toString().trim();

                                if (height1Numero.getEditableText().toString().trim().isEmpty()) {
                                    //vacio

                                    resultGrande.setText(R.string.falta);
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else if (Integer.parseInt(altura) < 0) {//negativo1
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrande.setText(R.string.negativo);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else {


                                    if (!genero().equals(null)) {
                                        double resultado = BodyMass.WanderVael(Integer.parseInt(altura), genero());
                                        resultGrande.setText(resultado + " Kg");
                                        resultGrande.setVisibility(View.VISIBLE);
                                        resultGrandeGraella.setVisibility(View.INVISIBLE);
                                    }

                                }


                            }

                        });//boton


                        break;

                    case R.id.LorentzRadio://el id que devuelve es el nombre pero en int
                        grupoGenero.clearCheck();

                        heightV1.setVisibility(View.VISIBLE);
                        height1Numero.setClickable(true);
                        height1Numero.setEnabled(true);

                        waist.setVisibility(View.INVISIBLE);
                        waistNumero.setClickable(false);
                        waistNumero.setEnabled(false);

                        heightV2.setVisibility(View.INVISIBLE);
                        height2Numero.setClickable(false);
                        height2Numero.setEnabled(false);

                        age.setVisibility(View.VISIBLE);
                        ageNumero.setClickable(true);
                        ageNumero.setEnabled(true);

                        resultGrande.setVisibility(View.INVISIBLE);
                        resultGrandeGraella.setVisibility(View.INVISIBLE);

                        grupo2.setClickable(false);
                        grupoGenero.setClickable(true);
                        maleRadio.setChecked(true);

                        boton.setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {

                                String altura = height1Numero.getEditableText().toString().trim();
                                String edad = ageNumero.getEditableText().toString().trim();

                                if (height1Numero.getEditableText().toString().trim().isEmpty() || ageNumero.getEditableText().toString().trim().isEmpty()) {
                                    //vacio

                                    resultGrande.setText(R.string.falta);
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else if (Integer.parseInt(altura) < 0 || Integer.parseInt(edad) < 0) {//negativo
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrande.setText(R.string.negativo);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else {


                                    if (!genero().equals(null)) {
                                        double resultado = BodyMass.Lorentz(Integer.parseInt(altura), Integer.parseInt(edad), genero());
                                        resultGrande.setText(resultado + " Kg");
                                        resultGrande.setVisibility(View.VISIBLE);
                                        resultGrandeGraella.setVisibility(View.INVISIBLE);
                                    }

                                }


                            }

                        });//boton

                        break;

                    case R.id.creffRadio:
                        grupoGenero.clearCheck();

                        heightV1.setVisibility(View.INVISIBLE);
                        height1Numero.setClickable(false);
                        height1Numero.setEnabled(false);


                        waist.setVisibility(View.INVISIBLE);
                        waistNumero.setClickable(false);
                        waistNumero.setEnabled(false);


                        heightV2.setVisibility(View.VISIBLE);
                        height2Numero.setClickable(true);
                        height2Numero.setEnabled(true);


                        age.setVisibility(View.VISIBLE);
                        ageNumero.setClickable(true);
                        ageNumero.setEnabled(true);

                        resultGrande.setVisibility(View.INVISIBLE);
                        resultGrandeGraella.setVisibility(View.INVISIBLE);

                        grupoGenero.setClickable(false);
                        grupo2.setClickable(true);

                        smallRadio.setChecked(true);

                        boton.setOnClickListener(new View.OnClickListener() {
                            @Override
                            public void onClick(View view) {

                                String altura = height2Numero.getEditableText().toString().trim();
                                String edad = ageNumero.getEditableText().toString().trim();

                                if (height2Numero.getEditableText().toString().trim().isEmpty() || ageNumero.getEditableText().toString().trim().isEmpty()) {
                                    //vacio

                                    resultGrande.setText(R.string.falta);
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else if (Integer.parseInt(altura) < 0 || Integer.parseInt(edad) < 0) {//negativo1
                                    resultGrande.setVisibility(View.VISIBLE);
                                    resultGrande.setText(R.string.negativo);
                                    resultGrandeGraella.setVisibility(View.INVISIBLE);
                                } else {


                                    if (!morphology().equals(null)) {
                                        double resultado = BodyMass.Creff(Integer.parseInt(altura), Integer.parseInt(edad), morphology());
                                        resultGrande.setText(resultado + " kg");
                                        resultGrande.setVisibility(View.VISIBLE);
                                        resultGrandeGraella.setVisibility(View.INVISIBLE);
                                    }


                                }


                            }

                        });//boton


                        break;
                }

            }
        });

    }

    public GENDER genero() {
        int seleccionadoGenero = grupoGenero.getCheckedRadioButtonId();
        switch (seleccionadoGenero) {
            case R.id.maleRadio:
                return GENDER.MALE;

            case R.id.femaleRadio:
                return GENDER.FEMALE;

            default:
                return null;

        }
    }

    public MORPHOLOGY morphology() {
        int seleccionadoG2 = grupo2.getCheckedRadioButtonId();
        switch (seleccionadoG2) {
            case R.id.smallRadio:
                return MORPHOLOGY.SMALL;

            case R.id.mediumRadio:
                return MORPHOLOGY.MEDIUM;

            case R.id.broadRadio:
                return MORPHOLOGY.BROAD;

            default:
                return null;

        }
    }


}

File: app/src/main/java/com/example/edt73kotlin/MainActivity.kt
package com.example.edt73kotlin

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import android.widget.*
import com.google.android.material.textfield.TextInputEditText

class MainActivity : AppCompatActivity() {

    private lateinit var grupoTop: RadioGroup
    private lateinit var fatRadio: RadioButton
    private lateinit var vaelRadio: RadioButton
    private lateinit var lorentzRadio: RadioButton
    private lateinit var creffRadio: RadioButton

    private lateinit var grupoGenero: RadioGroup
    private lateinit var maleRadio: RadioButton
    private lateinit var femaleRadio: RadioButton

    private lateinit var grupo2: RadioGroup
    private lateinit var smallRadio: RadioButton
    private lateinit var mediumRadio: RadioButton
    private lateinit var broadRadio: RadioButton

    private lateinit var heightV1: LinearLayout
    private lateinit var height1Numero: TextInputEditText
    private lateinit var heightV2: LinearLayout
    private lateinit var height2Numero: TextInputEditText
    private lateinit var waist: LinearLayout
    private lateinit var waistNumero: TextInputEditText
    private lateinit var age: LinearLayout
    private lateinit var ageNumero: TextInputEditText

    private lateinit var resultGrande: TextView
    private lateinit var resultGrandeGraella: LinearLayout
    private lateinit var boton: Button


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        val bodyMass = BodyMass()

        grupoTop = findViewById(R.id.eleccionSuperior)
        fatRadio = findViewById(R.id.fatRadio)
        vaelRadio = findViewById(R.id.WanRadio)
        lorentzRadio = findViewById(R.id.LorentzRadio)
        creffRadio = findViewById(R.id.creffRadio)

        grupoGenero = findViewById(R.id.genero)
        maleRadio = findViewById(R.id.maleRadio)
        femaleRadio = findViewById(R.id.femaleRadio)

        grupo2 = findViewById(R.id.radio2)
        smallRadio = findViewById(R.id.smallRadio)
        mediumRadio = findViewById(R.id.mediumRadio)
        broadRadio = findViewById(R.id.broadRadio)

        heightV1 = findViewById(R.id.heightLayoutv1)
        heightV2 = findViewById(R.id.heightLayoutv2)
        waist = findViewById(R.id.waistLayout)
        age = findViewById(R.id.ageLinear)
        resultGrande = findViewById(R.id.resultadoGrande)
        resultGrandeGraella = findViewById(R.id.graella)

        height1Numero = findViewById(R.id.height11Numero)
        height2Numero = findViewById(R.id.height2Numero)
        waistNumero = findViewById(R.id.waistNumero)
        ageNumero = findViewById(R.id.ageNumero)

        maleRadio = findViewById(R.id.maleRadio)
        smallRadio = findViewById(R.id.smallRadio)

        boton = findViewById(R.id.boton)


        heightV1.visibility = View.INVISIBLE
        waist.visibility = View.INVISIBLE
        heightV2.visibility = View.INVISIBLE
        age.visibility = View.INVISIBLE
        boton.visibility = View.INVISIBLE


        grupoTop.setOnCheckedChangeListener(RadioGroup.OnCheckedChangeListener { group, checkedId ->

            val radioSeleccionado = grupoTop.checkedRadioButtonId

            when (radioSeleccionado) {
                R.id.fatRadio -> {
                    heightV1.visibility = View.VISIBLE
                    height1Numero.isClickable = true
                    height1Numero.isEnabled = true

                    waist.visibility = View.VISIBLE
                    waistNumero.isClickable = true
                    waistNumero.isEnabled = true


                    heightV2.visibility = View.INVISIBLE
                    height2Numero.isClickable = false
                    height2Numero.isEnabled = false

                    age.visibility = View.INVISIBLE
                    ageNumero.isClickable = false
                    ageNumero.isEnabled = false


                    grupo2.isClickable = false
                    grupoGenero.isClickable = true


                    boton.visibility = View.VISIBLE


                    boton.setOnClickListener {
                        val altura = height1Numero.editableText.toString()
                        val anchura = waistNumero.getEditableText().toString()
                        //numeros

                        if (altura.isEmpty() || anchura.isEmpty()) {
                            //vacio
                            resultGrande.setText(R.string.falta)
                            resultGrande.visibility = View.VISIBLE
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else if (altura.toInt() < 0 || anchura.toInt() < 0) { //negativo
                            resultGrande.visibility = View.VISIBLE
                            resultGrande.setText(R.string.negativo)
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else {

                            val resultado =
                                bodyMass.FatMass(altura.toInt(), anchura.toInt(), genero())
                            //NO USAR EL LET
                            //porque siempre recibe null aunque este seleccionado por defecto

                            if(resultado!=-1.0){
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText("$resultado %")
                                resultGrandeGraella.visibility = View.VISIBLE
                            }
                            else{
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText(R.string.faltaRadio)
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }



                        }

                    }
                }
                R.id.WanRadio -> {
                    grupoGenero.clearCheck()

                    heightV1.visibility = View.VISIBLE
                    height1Numero.isClickable = true
                    height1Numero.isEnabled = true


                    waist.visibility = View.INVISIBLE
                    waistNumero.isClickable = false
                    waistNumero.isEnabled = false

                    heightV2.visibility = View.INVISIBLE
                    height2Numero.isClickable = false
                    height2Numero.isEnabled = false

                    age.visibility = View.INVISIBLE
                    ageNumero.isClickable = false
                    ageNumero.isEnabled = false

                    resultGrande.visibility = View.INVISIBLE
                    resultGrandeGraella.visibility = View.INVISIBLE

                    grupo2.isClickable = false
                    grupoGenero.isClickable = true


                    boton.visibility = View.VISIBLE


                    boton.setOnClickListener {
                        val altura = height1Numero.editableText.toString()
                        //numeros

                        if (altura.isEmpty()) {
                            //vacio
                            resultGrande.setText(R.string.falta)
                            resultGrande.visibility = View.VISIBLE
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else if (altura.toInt() < 0) { //negativo
                            resultGrande.visibility = View.VISIBLE
                            resultGrande.setText(R.string.negativo)
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else {

                            val resultado =
                                bodyMass.WanderVael(altura.toInt(), genero())
                            //NO USAR EL LET
                            //porque siempre recibe null aunque este seleccionado por defecto

                            if(resultado!=-1.0){
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText("$resultado Kg")
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }
                            else{
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText(R.string.faltaRadio)
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }

                        }

                    }

                }
                R.id.LorentzRadio -> {

                    grupoGenero.clearCheck()

                    heightV1.visibility = View.VISIBLE
                    height1Numero.isClickable = true
                    height1Numero.isEnabled = true

                    waist.visibility = View.INVISIBLE
                    waistNumero.isClickable = false
                    waistNumero.isEnabled = false

                    heightV2.visibility = View.INVISIBLE
                    height2Numero.isClickable = false
                    height2Numero.isEnabled = false

                    age.visibility = View.VISIBLE
                    ageNumero.isClickable = true
                    ageNumero.isEnabled = true

                    boton.visibility = View.VISIBLE

                    resultGrande.visibility = View.INVISIBLE
                    resultGrandeGraella.visibility = View.INVISIBLE

                    grupo2.isClickable = false
                    grupoGenero.isClickable = true



                    boton.setOnClickListener {
                        val altura = height1Numero.editableText.toString()
                        val edad = ageNumero.editableText.toString()
                        //numeros

                        if (altura.isEmpty()||edad.isEmpty()) {
                            //vacio
                            resultGrande.setText(R.string.falta)
                            resultGrande.visibility = View.VISIBLE
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else if (altura.toInt() < 0 || edad.toInt() < 0) { //negativo
                            resultGrande.visibility = View.VISIBLE
                            resultGrande.setText(R.string.negativo)
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else {

                            val resultado =
                                bodyMass.Lorentz(altura.toInt(),edad.toInt(), genero())
                            //NO USAR EL LET
                            //porque siempre recibe null aunque este seleccionado por defecto

                            if(resultado!=-1.0){
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText("$resultado Kg")
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }
                            else{
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText(R.string.faltaRadio)
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }


                        }

                    }
                }
                R.id.creffRadio -> {

                    grupoGenero.clearCheck()

                    heightV1.visibility = View.INVISIBLE
                    height1Numero.isClickable = false
                    height1Numero.isEnabled = false


                    waist.visibility = View.INVISIBLE
                    waistNumero.isClickable = false
                    waistNumero.isEnabled = false


                    heightV2.visibility = View.VISIBLE
                    height2Numero.isClickable = true
                    height2Numero.isEnabled = true


                    age.visibility = View.VISIBLE
                    ageNumero.isClickable = true
                    ageNumero.isEnabled = true

                    boton.visibility = View.VISIBLE

                    resultGrande.visibility = View.INVISIBLE
                    resultGrandeGraella.visibility = View.INVISIBLE

                    grupoGenero.isClickable = false
                    grupo2.isClickable = true

                    boton.setOnClickListener {
                        val altura = height2Numero.editableText.toString()
                        val edad = ageNumero.editableText.toString()
                        //numeros

                        if (altura.isEmpty()||edad.isEmpty()) {
                            //vacio
                            resultGrande.setText(R.string.falta)
                            resultGrande.visibility = View.VISIBLE
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else if (altura.toInt() < 0 || edad.toInt() < 0) { //negativo
                            resultGrande.visibility = View.VISIBLE
                            resultGrande.setText(R.string.negativo)
                            resultGrandeGraella.visibility = View.INVISIBLE
                        } else {

                            val resultado =
                                bodyMass.Creff(altura.toInt(),edad.toInt(), morphology())
                            //NO USAR EL LET
                            //porque siempre recibe null aunque este seleccionado por defecto

                            if(resultado!=-1.0){
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText("$resultado Kg")
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }
                            else{
                                resultGrande.visibility = View.VISIBLE
                                resultGrande.setText(R.string.faltaRadio)
                                resultGrandeGraella.visibility = View.INVISIBLE
                            }


                        }

                    }

                }

            }

        })


    }//oncreate

    fun genero(): GENDER? {
        val seleccionadoGenero = grupoGenero.checkedRadioButtonId
        return when (seleccionadoGenero) {
            R.id.maleRadio -> GENDER.MALE
            R.id.femaleRadio-> GENDER.FEMALE
            else -> null
        }
    }

    fun morphology(): MORPHOLOGY? {
        val seleccionadoG2 = grupo2.checkedRadioButtonId
        return when (seleccionadoG2) {
            R.id.smallRadio -> MORPHOLOGY.SMALL
            R.id.mediumRadio -> MORPHOLOGY.MEDIUM
            R.id.broadRadio->MORPHOLOGY.BROAD
            else -> null
        }
    }
}